test/accel/IntegrationTestBase.cc

```diff
@@ -0,0 +1,411 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file accel/IntegrationTestBase.cc
+//---------------------------------------------------------------------------//
+#include "IntegrationTestBase.hh"
+
+#include <exception>
+#include <limits>
+#include <G4Threading.hh>
+#include <G4UserEventAction.hh>
+#include <G4UserRunAction.hh>
+#include <G4UserSteppingAction.hh>
+#include <G4UserTrackingAction.hh>
+#include <G4VSensitiveDetector.hh>
+#include <G4VUserActionInitialization.hh>
+#include <G4VUserDetectorConstruction.hh>
+#include <G4VUserPrimaryGeneratorAction.hh>
+#include <G4Version.hh>
+
+#if G4VERSION_NUMBER >= 1100
+#    include <G4RunManagerFactory.hh>
+#else
+#    include <G4MTRunManager.hh>
+#endif
+
+#include "corecel/io/Logger.hh"
+#include "corecel/io/ScopedTimeAndRedirect.hh"
+#include "corecel/io/StringUtils.hh"
+#include "corecel/sys/TypeDemangler.hh"
+#include "geocel/GeantUtils.hh"
+#include "geocel/ScopedGeantExceptionHandler.hh"
+#include "geocel/ScopedGeantLogger.hh"
+#include "geocel/UnitUtils.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/Units.hh"
+#include "celeritas/ext/EmPhysicsList.hh"
+#include "celeritas/ext/SimpleSensitiveDetector.hh"
+#include "celeritas/g4/DetectorConstruction.hh"
+#include "celeritas/inp/Events.hh"
+#include "celeritas/phys/PDGNumber.hh"
+#include "accel/AlongStepFactory.hh"
+#include "accel/PGPrimaryGeneratorAction.hh"
+#include "accel/SetupOptions.hh"
+
+#include "PersistentSP.hh"
+
+namespace celeritas
+{
+namespace test
+{
+namespace
+{
+//---------------------------------------------------------------------------//
+std::string thread_label()
+{
+    if (int thread = G4Threading::G4GetThreadId(); thread >= 0)
+    {
+        return std::to_string(thread);
+    }
+    return "M";
```

Comment:
I think this is easier to read:

```suggestion
    return G4Threading::IsMasterThread() ? "M" : std::to_string(G4Threading::G4GetThreadId());
```